feat: configure Envoy to compress additional response types

darora opened the pull request:
Adds a number of Postgrest response types for compression by Envoy.

istarkov commented:
I can confirm that it works now, given that these changes have been applied to our Supabase project. 

However, I noticed that all other responses previously used Brotli compression by default, but now they are using gzip. 
[Support Ticket ID: 3125882163 / Discussion](https://github.com/orgs/supabase/discussions/28757)

_It’s not a big deal, but Brotli is more commonly used than gzip in our infrastructure._
_Anyway, we’ve excluded the use of .single() in our code, so we’ll be able to compare the difference between gzip and Brotli using our stats, if it exists and is statistically significant._

darora committed:
feat: configure Envoy to compress additional response types

Adds a number of Postgrest response types for compression by Envoy, enables gzip and brotli compression, and defaults to `br` compression (e.g. for `Accept-Encoding: *`).

darora commented:
We're looking into updating out Cloudflare configuration to prefer using Brotli. In the meantime, we'll be merging this in to have the docker images updated.
